### PR TITLE
Improve testing

### PR DESCRIPTION
**Pull request by Jamim**
Hello @nir0s,

I believe that it would be nice to have tests for actual and upcoming Python versions.

These changes also:
 - declare support for Python 3.7 and 3.8 at `setup.py`
 - remove Python 3.3 from `tox`'s `envlist`
 - remove [deprecated](https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration) `sudo` option from `.travis.yml`
 - fix typos

Best regards!

### CODE REVIEW

File: tests/test_distro.py [hunk 1]
```diff
@@ -1618,7 +1618,7 @@ def _bad_os_listdir(path='.'):
     raise OSError()
 
 
-@pytest.mark.skipIf(not IS_LINUX, reason='Irrelevant on non-linx')
+@pytest.mark.skipif(not IS_LINUX, reason='Irrelevant on non-linx')
```
Comment (GhostofGoes) [inline tests/test_distro.py]: Might as well fix the "linux" typo while you're at it 😄

Comment (Jamim) [reply tests/test_distro.py]: It's done. Thank you!

**Comment by nir0s** [comment]
@Jamim LGTM. Please rebase on top of master.

**Comment by Jamim** [comment]
Hello @nir0s,

> Please rebase on top of master.

It's done.

**Comment by Jamim** [comment]
It looks like Codecov checks got stuck for some reason.

**Comment by Jamim** [comment]
ping @nir0s

**Pull request merged**